src/NServiceBus.Core/Routing/Legacy/ReadyMessageSender.cs

```diff
@@ -66,7 +67,7 @@ public void MessageProcessed(Dictionary<string, string> headers)
             }
             else
             {
-                SendReadyMessage(1, false);
+                await SendReadyMessage(1, false).ConfigureAwait(false);
```

Comment:
Here there would be an opportunity to

`return SendReadyMessage`

and in the other exit points in the method

`return TaskEx.CompletedTask`

What do you think?